src/overlays/actors/ovl_En_Dinofos/z_en_dinofos.c

```diff
@@ -803,261 +825,266 @@ void func_8089BD28(EnDinofos* this, PlayState* play) {
     }
 
     this->actor.speed = 0.0f;
-    func_8089AD70(this);
+    EnDinofos_EnableBumperCollision(this);
     Actor_PlaySfx(&this->actor, NA_SE_EN_BOMCHU_WALK);
-    this->actionFunc = func_8089C44C;
+    this->actionFunc = EnDinofos_Land;
 }
 
-void func_8089C44C(EnDinofos* this, PlayState* play) {
+void EnDinofos_Land(EnDinofos* this, PlayState* play) {
     if (SkelAnime_Update(&this->skelAnime)) {
-        if ((this->unk_290 == 1) && (this->actor.xzDistToPlayer < 280.0f) && func_8089A968(this) &&
-            (Rand_ZeroOne() < 0.6f)) {
-            func_8089CA14(this);
+        if ((this->isJumpingBackward == true) && (this->actor.xzDistToPlayer < 280.0f) &&
+            EnDinofos_IsFacingPlayer(this) && (Rand_ZeroOne() < 0.6f)) {
+            EnDinofos_SetupStartBreatheFire(this);
         } else {
-            func_8089A9B0(this, play);
+            EnDinofos_ChooseAction(this, play);
         }
     }
 }
 
-void func_8089C4F8(EnDinofos* this) {
+void EnDinofos_SetupSlashFromGround(EnDinofos* this) {
     Animation_PlayOnce(&this->skelAnime, &gDinolfosAttackAnim);
-    this->colliderQuad.base.atFlags &= ~AT_BOUNCED;
+    this->knifeCollider.base.atFlags &= ~AT_BOUNCED;
     Actor_PlaySfx(&this->actor, NA_SE_EN_RIZA_CRY);
-    this->unk_290 = 0;
-    this->unk_292 = -1;
+    this->attackTimer = 0;
+    this->timer2 = -1;
     this->actor.speed = 0.0f;
     this->actor.world.rot.y = this->actor.shape.rot.y;
-    this->actionFunc = func_8089C56C;
+    this->actionFunc = EnDinofos_SlashFromGround;
 }
 
-void func_8089C56C(EnDinofos* this, PlayState* play) {
-    this->unk_290++;
+void EnDinofos_SlashFromGround(EnDinofos* this, PlayState* play) {
+    this->attackTimer++;
     Math_ScaledStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 0x800);
     this->actor.world.rot.y = this->actor.shape.rot.y;
     if (SkelAnime_Update(&this->skelAnime)) {
-        EffectBlure_AddSpace(Effect_GetByIndex(this->unk_2A0));
-        this->colliderQuad.base.atFlags &= ~AT_ON;
-        func_8089A9B0(this, play);
+        EffectBlure_AddSpace(Effect_GetByIndex(this->effectIndex));
+        this->knifeCollider.base.atFlags &= ~AT_ON;
+        EnDinofos_ChooseAction(this, play);
     } else if (Animation_OnFrame(&this->skelAnime, 7.0f)) {
-        this->colliderQuad.base.atFlags |= AT_ON;
-        func_8089AD70(this);
+        this->knifeCollider.base.atFlags |= AT_ON;
+        EnDinofos_EnableBumperCollision(this);
     } else if (Animation_OnFrame(&this->skelAnime, 13.0f)) {
-        this->colliderQuad.base.atFlags &= ~AT_ON;
+        this->knifeCollider.base.atFlags &= ~AT_ON;
     } else if (Animation_OnFrame(&this->skelAnime, 11.0f)) {
         Actor_PlaySfx(&this->actor, NA_SE_EN_RIZA_ATTACK);
     } else if (this->skelAnime.curFrame < 7.0f) {
-        func_8089AE00(this, play);
+        EnDinofos_Dodge(this, play);
     }
 }
 
-void func_8089C690(EnDinofos* this) {
+void EnDinofos_SetupRecoilFromBlockedSlash(EnDinofos* this) {
```

Comment:
```suggestion
void EnDinofos_SetupRecoil(EnDinofos* this) {
```
There is only one type of recoil so not sure we really need to add `FromBlockedSlash` here.